rust/windows-client/src-tauri/src/client/ipc.rs

```diff
@@ -271,6 +402,45 @@ impl Drop for SubcommandChild {
     }
 }
 
+/// Uses a Windows job object to kill child processes when the parent exits
+pub(crate) struct LeakGuard {
+    job_object: HANDLE,
+}
+
+impl LeakGuard {
+    pub fn new() -> Result<Self> {
+        let job_object = unsafe { CreateJobObjectA(None, None) }?;
+
+        let mut jeli = JOBOBJECT_EXTENDED_LIMIT_INFORMATION::default();
+        jeli.BasicLimitInformation.LimitFlags = JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE;
+        // SAFETY: Windows shouldn't hang on to `jeli`. I'm not sure why this is unsafe.
```

Comment:
All FFI is unsafe by default

Reply:
Yeah but the `windows` crate is supposed to have safe wrappers around some of the unsafe functions from `windows-sys`. Like here the XML document is safe https://github.com/microsoft/windows-rs/blob/master/crates/samples/windows/xml/src/main.rs

I guess they are making those safe wrappers by hand or something. For many of the new APIs I used the last week or two, they're only offering the auto-generated unsafe functions that take pointers.